### PR TITLE
Enhance error message 

### PR DESCRIPTION
**Pull request by Deren-Liao**
When Remote PowerShell fails to connect due to credential error etc, currently, it shows error with internal PowerShell commands etc.  It confuses user. 
This PR is to show some generic error message.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/AttachDebuggerDialog/AttachDebuggerSettingsStore.cs [hunk 1]
```diff
@@ -35,20 +35,21 @@ internal static class AttachDebuggerSettingsStore
...
         {
+            List<InstanceDefaultUser> results = null;
             if (File.Exists(DefaultUsersSettingsFullPath))
             {
                 string jsonText = File.ReadAllText(DefaultUsersSettingsFullPath);
                 if (!string.IsNullOrWhiteSpace(jsonText))
                 {
                     try
                     {
-                        return JsonConvert.DeserializeObject<List<InstanceDefaultUser>>(jsonText);
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/AttachDebuggerDialog/AttachDebuggerSettingsStore.cs]: It is purely a matter of opinion, but I prefer the way the function was written originally. The inner return was the focus of the method, and would not be missed when the function is read in the future.

File: GoogleCloudExtension/GoogleCloudExtension/AttachDebuggerDialog/AttachDebuggerFirewallPort.cs [hunk 2]
```diff
@@ -116,13 +116,14 @@ public int WaitForFirewallRuleTimeInSeconds()
...
-                    if (rule.TargetTags?.Contains(portTag) ?? false)
+                    if (rule?.TargetTags?.Contains(portTag) ?? false)
                     {
                         continue;   // Skip, rules does not contain the tag.
                     }
                     foreach (var allowed in rule.Allowed)
                     {
-                        if (allowed.IPProtocol == "tcp" && allowed.Ports.Any(y => y == PortInfo.Port.ToString()))
+                        if (allowed?.IPProtocol == "tcp" && 
+                            (allowed?.Ports?.Any(y => y == PortInfo.Port.ToString()) ?? false))
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/AttachDebuggerDialog/AttachDebuggerFirewallPort.cs]: Is this fixing an error you discovered?

Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/AttachDebuggerDialog/AttachDebuggerFirewallPort.cs]: No, I reviewed code and see this potentially throws null reference error.

File: GoogleCloudExtension/GoogleCloudExtension/AttachDebuggerDialog/AttachDebuggerSettingsStore.cs (hunk 1 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/AttachDebuggerDialog/AttachDebuggerSettingsStore.cs]: I am worried DeserializeObject may return null in some extreme cases.

**Comment by Deren-Liao** [comment]
PTAL.

**Pull request merged**